docs/deployment.md

```diff
@@ -0,0 +1,34 @@
+# Architecture-Overview
+
+![architecture-overview](images/ci_cd_github_big_picture_public.drawio.png)
+_The picture shows the architecture overview. The steps are explaind in the following passages_
+
+# Source Repo
+
+The top left from the architecture-overview is explaint in [link](./workflows.md).
+
+# Helm Chart
+
+![architekture-external-helm-chart](images/external-helm-chart.drawio.png)
+
+The top right from the architecture-overview is explaint in the following.
+
+We provide a sample chart. https://github.com/it-at-m/helm-charts/tree/main/charts/sps-sample, which consists subcharts. Each subchart is a different module for example frontend, backend, eai. Each module was created with `helm create` and we made some adaptions. The Chart.yml includes the external dependency of the [RefArch Api Gateway Helm Chart](https://github.com/it-at-m/helm-charts/tree/main/charts/refarch-gateway). The main config is done in the values.yml of the parent chart. In the sections refarch-gateway, frontend, backend you can configure the subcharts.
+
+# Deployment
+
+The bottom part from the architecture-overview is expliant in the following.
+
+At the end of a sprint a developer builds a release in the source code project. It needs to be done for each folder frontend/backend/eai separately. Then the devops engineer needs to update the chart-version and app-version of the helm chart in the Chart.yaml.
+
+Internal the quay syncs the docker images with the external github image registry. The github api has a pull limit so we need to cache the images internally. The configuration is done by the openshift team.
+
+![architekture-internal-helm-chart](images/internal-chart.drawio.png)
+
+The internal gitlab repo you can config your application, for example the trust store, ingress (openshift route), single sign on, database config, environment configs. The most config is comment out, so you need to comment it in. Here you see an example https://git.muenchen.de/ccse/cicd/sps-github
```

Comment:
Auf git.muenchen.de kann man nicht zugreifen von außen. Diese Verlinkung macht keinen Sinn.

Reply:
das ist halt auch blöd zwei Systeme Github/Gitlab zu Nutzen. Das Projekt auf Github zu Veröffentlichen macht auch keine Sinn, weil das interessante ist die Gitlab-Pipeline. Dann gehen wir zu Opencode.

Reply:
ich habe auch keine Probleme den internen Proxy-Adresse oder die Openshift API-Adresse zu veröffentlichen. Von außen kommt man da nicht drauf.

Reply:
> das ist halt auch blöd zwei Systeme Github/Gitlab zu Nutzen. Das Projekt auf Github zu Veröffentlichen macht auch keine Sinn, weil das interessante ist die Gitlab-Pipeline. Dann gehen wir zu Opencode.

Du kannst mal Klaus fragen, ob es in der OSS-Doku irgendwo steht, warum wir extern GitHub verwenden und nicht OpenCode. Dann könnten wir darauf verweisen. Ansonsten könnten wir den Grund hier natürlich auch reinschreiben, wenn es Sinn macht.

Reply:
> ich habe auch keine Probleme die internen Proxy-Adresse oder die Openshift API-Adresse zu veröffentlichen. Von außen kommt man da nicht drauf.

Das mag schon sein, aber ich finde Links im Internet, die dann ins Leere laufen und mir eine Fehlerseite anzeigen (DU hast da aber keinen Zugriff drauf) nervig und unprofessioniell.

Reply:
wenn ich jetzt was schreibe wie man es offen nutzt. dann geht es wieder intern nicht, wegen proxy, image nicht verfügbar etc.

Reply:
> Du kannst mal Klaus fragen, ob es in der OSS-Doku irgendwo steht, warum wir extern GitHub verwenden und nicht OpenCode.

> Wir richten uns nach der Community und nutzen ebenso GitHub für Public Code der Landeshauptstadt München. Durch die Codeverwaltung auf GitHub bieten wir eine niederschwellige Teilhabe für alle an.

https://opensource.muenchen.de/de/repositories.html#code

Reply:
> Here you see an example https://git.muenchen.de/ccse/cicd/sps-github

@ejcsid ich bin da bei dir das es blöd ist wenn man (nach extern) tote links im internet veröffentlicht. Aber in diesem Fall beschreiben wir ja wie wir den "Brückenkopf" wie freie Software und freie Container zu uns on-prem kommen.

Daher fände ich sogar @hupling Lösung vertretbar (mit einer Klammer dahinter)

Schön wäre es wenn wir ganz https://git.muenchen.de/ccse/cicd/sps-github/ raus mirroren (da wäre en mirror mal vertretbar;)

Opencode würde uns da nicht helfen, denn wir hätten immer ein internes gitlab für die interen CICD.

Reply:
> Schön wäre es wenn wir ganz https://git.muenchen.de/ccse/cicd/sps-github/ raus mirroren (da wäre en mirror mal vertretbar;)

Dann machen wir doch das und verlinken darauf. @hupling dann kannst du in der Doku erstmal rein schreiben, dass ein Beispiel noch kommt.